README.md

```diff
@@ -164,6 +164,7 @@ FLAGS:
     -i, --insecure               Proceed for server connections considered insecure (invalid TLS)
     -n, --no-progress            Do not show progress bar. This is recommended for non-interactive shells (e.g. for
                                  continuous integration)
+        --prefer-https           When HTTPS is available, treat HTTP links as errors
```

Comment:
I think that some people will have quite different intuition for the semantics of "prefer HTTPS".

To me, the behavior introduced in this PR is more like "enforce HTTPS", e.g. "error out in case HTTPS is not available".

"Prefer HTTPS" would be something between the lines of "try HTTPS first, but if HTTP is the only option, then use that instead".

Maybe we should leave the prefer vs enforce part to the user? E.g. something like:

`--https-mode=enforce` and `--https-mode=prefer`?

When unspecified or specified as a default (`ignore`?) it would not change the current behavior. This way, we'd have maximum flexibility, and unambiguous semantics.

Reply:
I will make a new commit and name it `require-https`